src/Agent.Plugins/GitSourceProvider.cs

```diff
@@ -849,6 +845,18 @@ public string GenerateAuthHeader(AgentTaskPluginExecutionContext executionContex
                 throw new InvalidOperationException($"Git fetch failed with exit code: {exitCode_fetch}");
             }
 
+            // If checking out by commit, explicity fetch it
+            // The branch with the commit may be pointing to a new merge commit, so this ensures it exists locally
```

Comment:
Does this sound like an accurate comment to add here for our future selves?

"This is done as a separate fetch rather than adding an additional refspec on the proceeding fetch so as not to potentially override any default refspecs picked up from the user's gitconfig when fetching origin"

Reply:
Good idea i'll leave that for future self